### PR TITLE
PEP 670: Rearrange and reword the PEP

### PR DESCRIPTION
**Pull request by encukou**
@vstinner & @erlend-aasland, this should make the PEP more understandable & complete

### CODE REVIEW

File: pep-0670.rst [hunk 1]
```diff
@@ -38,202 +34,132 @@ to maintain.
...
-The following macros should not be converted:
+The following macros will not be converted:
+
+* Object-like macros (i.e. those which don't need parentheses and
+  arguments). For example:
+
+  * Empty macros. Example: ``#define Py_HAVE_CONDVAR``.
+  * Macros only defining a value, even if a constant with a well defined
+    type would be better. Example: ``#define METH_VARARGS 0x0001``.
+  * Py_PYTHON_H
```
Comment (vstinner) [inline pep-0670.rst:85]: Just add Py_PYTHON_H as another example on the first item "Empty macros. Example: #define Py_HAVE_CONDVAR.", no?

File: pep-0670.rst [hunk 2]
```diff
@@ -527,18 +455,19 @@ using them as l-value in assignment. It is an incompatible change made
 on purpose: see `bpo-39573 <https://bugs.python.org/issue39573>`_ for
 the rationale.
 
-This PEP does not convert macros which can be used as l-value to avoid
-introducing incompatible changes.
+This PEP does propose converting macros which can be used as l-value
```
Comment (vstinner) [inline pep-0670.rst:458]:
```suggestion
This PEP does not propose converting macros which can be used as l-value
```

File: pep-0670.rst (hunk 1 above)
Comment (encukou) [reply pep-0670.rst:85]:
```suggestion
```
“used for definitions rather than behavior” is a better reason to exclude this one, IMO. It does fit both :)

**Comment by vstinner** [comment]
Please wait for Erlend's (@erlend-aasland) approval, since he is one of two authors of this PEP :-) cc @erlend-aasland

File: pep-0670.rst [hunk 3]
```diff
@@ -12,19 +12,15 @@ Python-Version: 3.11
...
+Macros in the C API will be converted to static inline functions or
+regular functions. This will help avoid macro pitfalls in C/C++, and
+make the functions usable from other programming languages.
 
-Convert macros and static inline functions to regular functions to make
-them usable by Python extensions which cannot use macros or static
-inline functions, like extensions written in a programming languages
-other than C or C++.
+To avoid compiler warnings, function arguments of pointer types
+will be cast to appropriate types using additional macros.
```
[the remark this answers is not in the document]
Comment (vstinner) [reply pep-0670.rst:20]: Yep, but the parenthesis is not needed. It's an important information.

**Review by erlend-aasland — APPROVED** [review]
AFAICS, the major part of this PR is a rearrangement of some of the sections. I'm fine with that. The rewordings are ok to me as well.

Thanks Petr and Victor.

**Pull request merged**